### PR TITLE
Browse dept

### PR DESCRIPTION
**Pull request by lfarrell**
Search within a collection by department, if the collection has departments associated with it.

### CODE REVIEW

File: access/src/main/java/edu/unc/lib/dl/ui/controller/BrowseDepartmentsController.java [hunk 1]
```diff
@@ -17,28 +17,44 @@
...
-	public String handleRequest(Model model){
-		FacetFieldObject deptField = queryLayer.getDepartmentList(GroupsThreadStore.getGroups());
+	@RequestMapping(value = "/{pid}", method = RequestMethod.GET)
+	public String handleRequest(@PathVariable("pid") String pid, Model model){
+		SearchResultResponse result;
+		
+		if (pid != null) {
+			result = queryLayer.getDepartmentList(GroupsThreadStore.getGroups(), pid);
+		} else {
+		    result = queryLayer.getDepartmentList(GroupsThreadStore.getGroups(), null);
```
Comment (bbpennel) [inline access/src/main/java/edu/unc/lib/dl/ui/controller/BrowseDepartmentsController.java]: Switch indent to a tab

File: access/src/main/java/edu/unc/lib/dl/ui/controller/BrowseDepartmentsController.java [hunk 2]
```diff
@@ -17,28 +17,44 @@
...
+		}		
 
-		if (deptField != null)
-			model.addAttribute("departmentFacets", deptField);
+		if (result != null) {
+			model.addAttribute("departmentFacets", result.getFacetFields().get(0));
+		    model.addAttribute("container", result.getSelectedContainer());
```
Comment (bbpennel) [inline access/src/main/java/edu/unc/lib/dl/ui/controller/BrowseDepartmentsController.java]: Switch indent to a tab

File: .gitignore [hunk 3]
```diff
@@ -12,4 +12,5 @@
 .DS_Store
 /ssh_config
 /static.tar.gz
-/puppet.tar.gz
\ No newline at end of file
+/puppet.tar.gz
+.idea
```
Comment (daines) [inline .gitignore]: Please include this change in a separate commit.

File: access/src/main/webapp/WEB-INF/jsp/common/searchBox.jsp [hunk 4]
```diff
@@ -76,7 +76,7 @@
 		<h2 class="fpsearch_browse_separator italics">or</h2>
 		<div class="fpsearch_browse_links">
 			<a href="collections">Browse the Collections</a><br/>
-			<a href="browseDepartments">Browse Departments</a>
+			<a href="browse/depts/">Browse Departments</a>
```
Comment (daines) [inline access/src/main/webapp/WEB-INF/jsp/common/searchBox.jsp]: Typo? "depts"

Also, is this code ever reached?

File: access/src/main/java/edu/unc/lib/dl/ui/controller/AdvancedSearchFormController.java [hunk 5]
```diff
@@ -49,8 +49,8 @@ public String handleRequest(Model model, HttpServletRequest request){
 			SearchResultResponse collectionResultResponse = queryLayer.getCollectionList(accessGroups);
 			model.addAttribute("collectionList", collectionResultResponse.getResultList());
 			
-			FacetFieldObject departmentFacetFieldObject = queryLayer.getDepartmentList(accessGroups);
-			model.addAttribute("departmentList", departmentFacetFieldObject.getValues());
+			SearchResultResponse departmentFieldObject = queryLayer.getDepartmentList(accessGroups, null);
+			model.addAttribute("departmentList", departmentFieldObject);
```
Comment (daines) [inline access/src/main/java/edu/unc/lib/dl/ui/controller/AdvancedSearchFormController.java]: This appears to break the advanced search page somehow -- at least on my VM. It works in master.

This is the exception I got: https://gitlab.lib.unc.edu/snippets/47

File: access/src/main/webapp/WEB-INF/jsp/common/searchBox.jsp (hunk 4 above)
Comment (bbpennel) [inline access/src/main/webapp/WEB-INF/jsp/common/searchBox.jsp]: no, its not reached, we could remove it at this point.

**Pull request merged**